include/gatekeeper_lls.h

```diff
@@ -19,6 +19,179 @@
 #ifndef _GATEKEEPER_LLS_H_
 #define _GATEKEEPER_LLS_H_
 
-int run_lls(void);
+#include <rte_timer.h>
+
+#include "gatekeeper_mailbox.h"
+#include "gatekeeper_net.h"
+
+/*
+ * Maximum key length (in bytes) for an LLS entry. It should be set
+ * to the maximum key length needed. Currently, this is set to 16 for
+ * the number of bytes in an IPv6 address.
+ */
+#define LLS_MAX_KEY_LEN 16
+
+/* Number of records that a LLS cache can hold. */
+#define LLS_CACHE_RECORDS 1024
+
+struct lls_entry {
+	/* Ethernet address of this entry. */
+	struct ether_addr ha;
+
+	/* Port on which this entry exists. */
+	uint8_t           port_id;
+
+	/* Whether this entry has been marked as stale. */
+	int               stale;
+
+	/* IP address for this resolution, in network ordering. */
+	uint8_t           ip_be[LLS_MAX_KEY_LEN];
+};
+
+/* Format of callback function for requesting LLS entries. */
+typedef void (*lls_update_cb)(struct lls_entry *entry, void *arg);
+
+enum lls_request_ty {
+	/* Do a one-time resolution; do not receive further updates. */
+	LLS_REQUEST_RESOLVE,
+	/* Request to receive a resolution and any updates for this record. */
+	LLS_REQUEST_HOLD,
+	/* Remove a previously-submitted request. */
+	LLS_REQUEST_PUT,
```

Comment:
I see. It seems the LLS block must notify the block via the callback when the subscription is removed. In my next rebase, the callback will include a message type as well.